src/arrays/referencing_array_and_its_elements/hard/q5/MainTest.java

```diff
@@ -0,0 +1,95 @@
+package arrays.referencing_array_and_its_elements.hard.q5;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.utils.ArrayUtil.arrayToInput;
+import static global.utils.ArrayUtil.generateRandomArray;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+    public static double pivot = 60.0;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "76.9 50.8 68.4 95.7 88.6 81.2 89.5 74.9 12.76 64.0";
+        return new Clause[]{
+                new StringLiteral("Enter " + n + " double numbers:"),
```

Comment:
Add the optional here like I talked about in https://github.com/canvas-gamification/canvas-gamification-junit-tests/pull/461